rootchain/sampool/bucket.go

```diff
@@ -0,0 +1,63 @@
+package sampool
+
+import (
+	"github.com/0xPolygon/polygon-edge/rootchain"
+	"github.com/0xPolygon/polygon-edge/types"
+)
+
+type samSet struct {
+	messages   []rootchain.SAM
+	signatures map[string]bool
+}
+
+func newUniqueSAMs() samSet {
+	return samSet{
+		messages:   make([]rootchain.SAM, 0),
+		signatures: make(map[string]bool),
+	}
+}
+
+func (s *samSet) add(msg rootchain.SAM) {
+	strSignature := string(msg.Signature)
+
+	if s.signatures[strSignature] {
+		return
+	}
+
+	s.messages = append(s.messages, msg)
+	s.signatures[strSignature] = true
+}
+
+func (s *samSet) get() []rootchain.SAM {
```

Comment:
I think `Messages` sounds better because I don't know what `get` returns from the name `get` and `samSet`.

Reply:
That's why I always look at the return type.

[Renamed](https://github.com/0xPolygon/polygon-edge/pull/718/commits/0e4b695ce070ddd52a5a2f7840d08c09cfd02c78)